File: graphql-dgs-codegen-core/src/integTest/java/com/netflix/graphql/dgs/codegen/java/testcases/dataclasses/dataClassWithNonNullableComplexType/expected/types/MyType.java
package com.netflix.graphql.dgs.codegen.java.testcases.dataclasses.dataClassWithNonNullableComplexType.expected.types;

import java.lang.Object;
import java.lang.Override;
import java.lang.String;
import java.util.Objects;

public class MyType {
  private OtherType other;

  public MyType() {
  }

  public MyType(OtherType other) {
    this.other = other;
  }

  public OtherType getOther() {
    return other;
  }

  public void setOther(OtherType other) {
    this.other = other;
  }

  @Override
  public String toString() {
    return "MyType{other='" + other + "'}";
  }

  @Override
  public boolean equals(Object o) {
    if (this == o) return true;
    if (o == null || getClass() != o.getClass()) return false;
    MyType that = (MyType) o;
    return Objects.equals(other, that.other);
  }

  @Override
  public int hashCode() {
    return Objects.hash(other);
  }

  public static Builder newBuilder() {
    return new Builder();
  }

  public static class Builder {
    private OtherType other;

    public MyType build() {
      MyType result = new MyType();
      result.other = this.other;
      return result;
    }

    public Builder other(OtherType other) {
      this.other = other;
      return this;
    }
  }
}

File: graphql-dgs-codegen-core/src/integTest/kotlin/com/netflix/graphql/dgs/codegen/cases/dataClassWithNonNullableComplexType/expected/types/MyType.kt
package com.netflix.graphql.dgs.codegen.cases.dataClassWithNonNullableComplexType.expected.types

import com.fasterxml.jackson.`annotation`.JsonIgnoreProperties
import com.fasterxml.jackson.`annotation`.JsonProperty
import com.fasterxml.jackson.`annotation`.JsonTypeInfo
import com.fasterxml.jackson.databind.`annotation`.JsonDeserialize
import com.fasterxml.jackson.databind.`annotation`.JsonPOJOBuilder
import com.netflix.graphql.dgs.codegen.cases.dataClassWithNonNullableComplexType.expected.Generated
import java.lang.IllegalStateException
import kotlin.Any
import kotlin.Boolean
import kotlin.Int
import kotlin.String
import kotlin.collections.List
import kotlin.jvm.JvmName

@Generated
@JsonTypeInfo(use = JsonTypeInfo.Id.NONE)
@JsonDeserialize(builder = MyType.Builder::class)
public class MyType(
  other: () -> OtherType = otherDefault,
) {
  private val __other: () -> OtherType = other

  @get:JvmName("getOther")
  public val other: OtherType
    get() = __other.invoke()

  private fun `__$fieldValues`(): List<Any?> = listOf(
      if (__other === otherDefault) otherDefault else other,
  )

  override fun equals(other: Any?): Boolean = this === other || (other is MyType &&
      `__$fieldValues`() == other.`__$fieldValues`())

  override fun hashCode(): Int = `__$fieldValues`().hashCode()

  private fun `__$fieldStrings`(): List<String> = listOfNotNull(
      if (__other === otherDefault) null else "other=" + other,
  )

  override fun toString(): String = `__$fieldStrings`().joinToString(prefix = "MyType(", postfix =
      ")")

  @Generated
  public companion object {
    private val otherDefault: () -> OtherType = 
        { throw IllegalStateException("Field `other` was not requested") }
  }

  @Generated
  @JsonPOJOBuilder
  @JsonIgnoreProperties("__typename")
  public class Builder {
    private var other: () -> OtherType = otherDefault

    @JsonProperty("other")
    public fun withOther(other: OtherType): Builder = this.apply {
      this.other = { other }
    }

    public fun build(): MyType = MyType(
      other = other,
    )
  }
}
